source/server/listener_impl.cc

```diff
@@ -168,25 +168,36 @@ void ListenSocketFactoryImpl::doFinalPreWorkerInit() {
     return;
   }
 
-  for (auto& socket : sockets_) {
-    const auto rc = socket->ioHandle().listen(tcp_backlog_size_);
-#ifndef WIN32
+  if (sockets_.empty()) {
```

Comment:
In what case can this be empty?

Reply:
I don't think it can be empty, just being defensive as possible. Should I just remove the check?

Reply:
If it can't be empty, switch to ASSERT seems fine.

Reply:
+1 this should never be empty.

Reply:
changed it to an assert in 5034e6e7bec0bb960b0a89e0ada66b94ebf62205